File: Fragment/app/src/main/java/com/example/fragment/MainActivity.java
package com.example.fragment;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "tag";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.i(TAG,"onCreatActivity");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG,"onStartActivity");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG,"onPauseactivity");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG,"onStopActivity");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i(TAG,"onDestroyActivity");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i(TAG,"onRestartActivity");
    }

    @Override
    protected void onResume() {
        super.onResume();
    }
}

File: FragmentStudying/app/src/main/java/com/example/fragmentstudying/MainActivity.kt
package com.example.fragmentstudying

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.fragment.app.FragmentManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() , OnClick {
    var myNames: List<Name> = listOf(Name("Milad"),Name("fghjjjb "),Name("cadaf"),Name("qwweqe"),Name("fff"),Name("88888"),Name("safhasf"))
    lateinit var recyclerView: RecyclerView
    lateinit var ad:NameAdabter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
      recyclerView = findViewById(R.id.main_rv)
        ad = NameAdabter(myNames , this )
       recyclerView.apply {
           adapter=ad
           layoutManager = LinearLayoutManager(this@MainActivity)
       }

    }

    override fun click(name: Name) {
        var bundle:Bundle = Bundle()
        bundle.putString("myName",name.n)
        var first:FisrtFragment = FisrtFragment()
        first.arguments = bundle
        supportFragmentManager.beginTransaction().replace(R.id.my_contaner,first).commit()
    }
}